src-jdbc4/net/sf/log4jdbc/PreparedStatementSpy.java

```diff
@@ -52,6 +52,8 @@ public class PreparedStatementSpy extends StatementSpy implements PreparedStatem
    */
   protected final List argTrace = new ArrayList();
 
+  private static final boolean showParams = Boolean.parseBoolean(System.getProperty("SQL_SHOW_PARAMS"));

```

Comment:
We should name it `LOG4JDBC_SQL_SHOW_PARAMS`.

Please ignore if the code already handles it this way, if not we should update to do it.
- Default should be false. We should not expect the variable to be defined. If the variable is not defined or is false, then it does not log params. The new default
- If it is defined and value is true, then we show params. Most likely in local and INT. 

Just thought of it because I don't remember if `parseBoolean` throws an exception for invalid values. Either way, we should the above word about the behavior in one of our documents.

Reply:
Have changed the prop name to be `LOG4JDBC_SQL_SHOW_PARAMS`.

The code handles exactly as you described: `false` if not specified and `true` if a value of the param is `true`.

`parseBoolean` does not throw exceptions because its implementation is `"true".equalsIgnoreCase(s);`. So, maybe it makes sense to throw an exception if param is present, but value is not a valid boolean literal. What do you think?

Reply:
I don't have a preference. I think specifically looking for a `true` value in the variable is simple, less work and robust enough; like you have to intentionally turn on the switch. Absence of the variable or any other value is false.